File: c/exs1/pset1/cash.c
#include <stdio.h>
#include <math.h>

int main() {
    printf("Quanto dinheiro estamos te devendo: ");
    float troco;
    int moedas = 0;

 
    while (1) {
        if (scanf("%f", &troco) == 1) {
            if (troco > 0.0) {
                break; 
            } else {
                printf("Valor deve ser positivo. Tente novamente: ");
            }
        } else {
            printf("Entrada inválida. Tente novamente: ");
            while (getchar() != '\n');  // Limpa o buffer de entrada
        }
    }
    
    int centavos = round(troco * 100);
    printf("%d\n",centavos);
    while(centavos >= 0.25){    
        if(centavos < 25){
            break;
        }
        centavos -= 25;
        moedas++;
            printf("%d\n",centavos);
    
    }
    while(centavos >= 0.10){
        if(centavos <10){
            break;
        }
        centavos -= 10;
        moedas++;
            printf("%d\n",centavos);
    }
    while(centavos >= 0.05){
        if(centavos < 5){
            break;
        }
        centavos -= 5;
        moedas++;
            printf("%d\n",centavos);
    }
       while (centavos >= 1) {
           if(centavos < 1){
            break;
        }
        centavos -= 1;
        moedas++;
            printf("%d\n",centavos);
    }

    printf("Número total de moedas: %d\n", moedas);

    return 0;
    
}

File: python/exs/cash.py
troco = float(input("Quanto dinheiro estamos lhe devendo: "))

while troco <= 0:
    troco = float(input("Numero invalido tente novamente: "))

moedas  = 0
centavos = int(round(troco * 100))

while centavos >= 25:

    centavos -= 25
    moedas+=1
    print(centavos)

while centavos >= 10:
    centavos -= 10
    moedas+=1
    print(centavos)

while centavos >= 5:
    centavos -= 5
    moedas+=1
    print(centavos)

while centavos >= 1:
    centavos -= 1
    moedas+=1
    print(centavos)

print("Número total de moedas: ", moedas)
